### PR TITLE
JAMES-2292 Bind MailQueueRoutes

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java [hunk 1]
```diff
@@ -42,6 +42,7 @@
 import org.apache.james.modules.server.DataRoutesModules;
 import org.apache.james.modules.server.ElasticSearchMetricReporterModule;
 import org.apache.james.modules.server.JMXServerModule;
+import org.apache.james.modules.server.MailQueueRoutesModule;
```
Comment (chibenwa) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java]: Did you forgot to commit MailQueueRoutesModule ? I can not see that file...

File: server/container/guice/cassandra-guice/pom.xml [hunk 2]
```diff
@@ -168,6 +168,10 @@
             <groupId>${project.groupId}</groupId>
             <artifactId>james-server-guice-webadmin-mailbox</artifactId>
         </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>james-server-guice-webadmin-mailqueue</artifactId>
```
Comment (chibenwa) [inline server/container/guice/cassandra-guice/pom.xml]: Where is that POM defined?

**Comment by aduprat** [comment]
test this please
```
Running org.apache.james.queue.memory.MemoryMailQueueTest
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Tests run: 61, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 9.666 sec <<< FAILURE! - in org.apache.james.queue.memory.MemoryMailQueueTest
delayShouldAtLeastBeTheOneSpecified()  Time elapsed: 1.009 sec  <<< FAILURE!
java.lang.AssertionError: 

Expecting:
 <1000L>
to be greater than:
 <1000L> 
	at org.apache.james.queue.api.DelayedMailQueueContract.delayShouldAtLeastBeTheOneSpecified(DelayedMailQueueContract.java:115)
```

File: server/protocols/webadmin/webadmin-mailqueue/src/main/java/org/apache/james/webadmin/routes/MailQueueRoutes.java [hunk 3]
```diff
@@ -74,8 +74,9 @@
     private final JsonTransformer jsonTransformer;
 
     @Inject
-    @VisibleForTesting MailQueueRoutes(MailQueueFactory<ManageableMailQueue> mailQueueFactory, JsonTransformer jsonTransformer) {
-        this.mailQueueFactory = mailQueueFactory;
+    @SuppressWarnings("unchecked")
+    @VisibleForTesting MailQueueRoutes(MailQueueFactory<?> mailQueueFactory, JsonTransformer jsonTransformer) {
+        this.mailQueueFactory = (MailQueueFactory<ManageableMailQueue>) mailQueueFactory;
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-mailqueue/src/main/java/org/apache/james/webadmin/routes/MailQueueRoutes.java]: It looks like the consequences of not using the right generics in injection (and probably not using TypeLiteral). It's not important for now but should be removed in the future.

Comment (aduprat) [reply server/protocols/webadmin/webadmin-mailqueue/src/main/java/org/apache/james/webadmin/routes/MailQueueRoutes.java]: I agree with you, the thing is that sometime we are waiting for a _MailQueueFactory<MailQueue>_ to be injected and other time for a _MailQueueFactory<ManageableMailQueue>_

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**